project/src/main/java/com/google/sps/utilities/DatabaseWrapper.java

```diff
@@ -25,14 +19,9 @@ public DatabaseWrapper(String instanceId, String databaseId) {
    * @param user the user to be inserted; user's ID field should not exist in DB
    */
   public void insertUser(User user) {
-    SpannerOptions options = SpannerOptions.newBuilder().build();
-    Spanner spanner = options.getService();
-    DatabaseId db = DatabaseId.of(options.getProjectId(), instanceId, databaseId);
-    DatabaseClient dbClient = spanner.getDatabaseClient(db);
-
     List<Mutation> mutations = getMutationsFromBuilder(newInsertBuilderFromUser(), user);
-    dbClient.write(mutations);
-    spanner.close();
+    databaseService.getDatabaseClient().write(mutations);
+    databaseService.getSpanner().close();
```

Comment:
No need to close the Spanner here?

This should be called when you turn down the whole server.

Reply:
Done. I see - thanks.